packages/compass-crud/src/utils/cancellable-queries.ts

```diff
@@ -119,23 +141,23 @@ export async function fetchShardingKeys(
 
   // close the cursor if the operation is aborted
   const abort = () => {
-    cursor.close();
+    void cursor.close();
   };
   signal.addEventListener('abort', abort, { once: true });
 
   let configDocs;
 
   try {
     configDocs = await raceWithAbort(cursor.toArray(), signal);
-  } catch (err) {
+  } catch (err: any) {
```

Comment:
Is it really `any` here or do we expect some kind of error format? when we check below for `err.message` we know that err is not undefined or something.

Reply:
So, yeah, we're definitely expecting that `err` is an `Error` instance here. That seems reasonable, because `signal` is under our control, and `cursor.toArray()` is a driver method, and the driver team is a team of good JS engineers who won’t write code that throws `undefined` or `null` as exceptions :)

It’s good that you point this out, though. I do think it’s much nicer how the aggregations package handles this, using an `Error` subclass with a proper `name` property. Since the code is mostly duplicated anyway and this PR might be a good opportunity to clean this up, I’ve decided to move this to `compass-utils` in 6cff480fe9be73c9d96394b3ffd227c031560a38, and then followed up with adding a bit more web compat in 548e098e663e80a9ba261251fe6c6bf50a11953c, feel free to take a look :)

Reply:
Nice! so we don't need any type here anymore since isCancelError expects any?

Reply:
The interesting part of the `isCancelError` type definition is the `error is ...` return type, which means it returns `true` if the error is a cancel/abort error and `false` otherwise, and TS can use that information at the call site (like here) to know "ah yes, the type was checked here".